### ISSUE TITLE
consider default output and temp directories

### ISSUE THREAD

**Issue by marisademeglio** [daisy/pipeline-gui#31]
"Maybe output and temp directories could have a default location, such as in the system temp directory? Or maybe not, just an idea. At least, the file dialog for output directories should probably open in the same location as the last selected output directory from the previous job; my guess is that users keep all their result files in the same location."

Can we detect when a job option is for an output or temp directory?

**Comment by josteinaj** [comment]
We have the `px:output="result|temp"` attribute in the XProc scripts. However, I think the framework handles these directories so that they are not exposed through the Web API at all. How that works when you use the Java API I have no idea.

**Comment by marisademeglio** [comment]
this will be a good future addition along with other user preferences

**Comment by marisademeglio** [comment]
yeah i thought the framework dealt with temporary directories too. maybe it does and exposing them in the GUI is redundant?

**Comment by bertfrees** [comment]
I don't think this is the case, see issue https://github.com/daisy/pipeline-gui/issues/64.

**Comment by bertfrees** [comment]
I'll check.

**Comment by bertfrees** [comment]
It seems indeed that the framework deals with this. Now that https://github.com/daisy/pipeline-gui/issues/64 is fixed, if you leave the field empty, the framework will fill it in for you. In theory we could indeed not expose these options, however that would mean some useful documentation is also skipped (see also my comment in https://github.com/daisy/pipeline-gui/issues/63).

**Comment by bertfrees** [comment]
For now I'm filling in the temporary directory fields but still exposing them.

**Comment by bertfrees** [comment]
Not exposing certain options has another drawback: the framework will automatically store result files in a generated directory which may not be desirable, for example because it makes opening the files trickier (see https://github.com/daisy/pipeline-gui/issues/70).
